generator/internal/connection_impl_generator.cc

```diff
@@ -119,39 +119,25 @@ class $connection_class_name$Impl
   // So, we have to fallback to `options_`.
   HeaderPrint(R"""(
  private:
-  std::unique_ptr<$product_namespace$::$retry_policy_name$> retry_policy() {
-    auto const& options = internal::CurrentOptions();
-    if (options.has<$product_namespace$::$retry_policy_name$Option>()) {
-      return options.get<$product_namespace$::$retry_policy_name$Option>()->clone();
-    }
-    return options_.get<$product_namespace$::$retry_policy_name$Option>()->clone();
+  static std::unique_ptr<$product_namespace$::$retry_policy_name$>
```

Comment:
optional nit: do we prefer to make this a standalone function in the anonymous namespace?

Reply:
Done.